gcp-client/src/main/java/com/google/graphite/platforms/plugin/client/util/ClientUtil.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.graphite.platforms.plugin.client.util;
+
+import com.google.common.collect.ImmutableList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/** A library of utility functions for common interactions with the GCP client libraries. */
+public class ClientUtil {
+
+  /**
+   * Given a list of GCP resources, filters the list using the provided filter and sorts according
+   * to the provided comparator.
+   *
+   * @param items A list of GCP resources produced by a GCP client library request.
+   * @param filter A predicate applied to objects in {@param items}. If true for a given object,
+   *     then that object will be kept in the final result.
+   * @param comparator Defines a comparison between any two objects in {@param items} used for
+   *     sorting the result.
+   * @param <T> The type of the list elements
+   * @return An {@link ImmutableList} which is empty if {@param items} is null or empty, and
+   *     otherwise is {@param items} filtered and sorted as described above.
+   */
+  public static <T> ImmutableList<T> processResourceList(
+      final List<T> items, final Predicate<T> filter, final Comparator<T> comparator) {
+    if (items == null) {
+      return ImmutableList.of();
+    }
+    return ImmutableList.copyOf(
+        items.stream().filter(filter).sorted(comparator).collect(Collectors.toList()));
+  }
+
+  /**
+   * Given a list of GCP resources, sorts according to the provided comparator.
+   *
+   * @param items A list of GCP resources produced by a GCP client library request.
+   * @param comparator Defines a comparison between any two objects in {@param items} used for
+   *     sorting the result.
+   * @param <T> The type of the list elements.
+   * @return An {@link ImmutableList} which is empty if {@param items} is null or empty, and
+   *     otherwise is {@param items} filtered and sorted as described above.
+   */
+  public static <T> ImmutableList<T> processResourceList(
```

Comment:
sortResourceList might be a more appropriate method name, since that's all this method seems to do.

Reply:
It does a bit more by handling the null input case nicely, which takes those repeated null checks out of the client code, and I thought it made sense to have an overload for this method without any filtering.